include/deal.II/lac/linear_operator.h

```diff
@@ -1111,6 +1111,112 @@ linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
   return return_op;
 }
 
+template <typename Range = Vector<double>,
+          typename Domain = Range,
+          typename Matrix>
+LinearOperator<Range, Domain>
+constraints_linear_operator(const ConstraintMatrix &, const Matrix &);
+
+/**
+ * @relates LinearOperator
+ *
+ * constraints_linear_operator takes a constraint matrix @p cm and an exemplar
+ * matrix @p (this matrix could be the system matrix).
+ * The result of this function is the LinearOperator associated to the constraint
+ * matrix.
+ *
+ * [ see M. S. Shephard:. Linear multipoint constraints applied via
+ * transformation as part of a direct stiffness assembly process., 1985.
+ * for more deatails ]
+ *
+ * @ingroup LAOperators
+ */
+template <  typename Range,
+            typename Domain,
+            typename Matrix >
+LinearOperator<Range, Domain>
+constraints_linear_operator(const ConstraintMatrix &cm, const Matrix &m)
+{
+  LinearOperator<Range, Domain> return_op = linear_operator<Range, Domain>(m);
+
+  return_op.vmult = [&cm](Range &v, const Domain &u)
+  {
+    for (auto i : v.locally_owned_elements())
+      {
+        if (cm.is_constrained(i))
+          {
+            v(i) = 0;
+            const std::vector< std::pair < types::global_dof_index, double > >
+            *entries = cm.get_constraint_entries (i);
+            for (types::global_dof_index j=0; j < entries->size(); ++j)
+              {
+                types::global_dof_index pos = (*entries)[j].first;
+                v(i) +=  u(pos) * (*entries)[j].second;
+              }
+          }
+        else
+          v(i) = u(i);
+      }
+  };
+
+  return_op.Tvmult = [&cm](Range &v, const Domain &u)
+  {
+    for (auto i : u.locally_owned_elements())
+      {
+        if (cm.is_constrained(i))
+          {
+            v(i)=0;
+            const std::vector< std::pair < types::global_dof_index, double > >
+            *entries = cm.get_constraint_entries (i);
+            for (types::global_dof_index j=0; j < entries->size(); ++j)
+              {
+                types::global_dof_index pos = (*entries)[j].first;
+                v(pos) += u(i) * (*entries)[j].second;
+              }
+          }
+        else
+          v(i)=u(i);
+
+      }
+  };
+
+  return return_op;
+}
+
+template <typename Range = Vector<double>,
+          typename Domain = Range,
+          typename Matrix>
+LinearOperator<Range, Domain>
+constrained_linear_operator(const ConstraintMatrix &, const Matrix &);
+
+/**
+ * @relates LinearOperator
+ *
+ * Let @p m be a system matrix and assume we are interested in this problem
+ * constrained with a constraint matrix @p cm.
+ * This function returns the LinearOperator that replace the system matrix and
+ * is associated to this problem.
```

Comment:
same here. I think it would be useful to explicitly spell out what operation this LinearOperator does.